### PR TITLE
Some more detailed tests for success needed for a project

### PR DESCRIPTION
**Pull request by hendrik-skubch**
Success for plans is calculated on the team level. E.g., A plan succeeds for all robots involved or does not succeed at all. In some cases a reaction is wanted if only the local agent completed a task in that plan.

### CODE REVIEW

File: alica_engine/include/engine/RunningPlan.h [hunk 1]
```diff
@@ -194,6 +194,8 @@ class RunningPlan : public std::enable_shared_from_this<RunningPlan>
     bool isAnyChildStatus(PlanStatus ps) const;
     bool areAllChildrenStatus(PlanStatus ps) const;
     bool isAnyChildTaskSuccessful() const;
+    bool AmISuccessful() const;
```
Comment (abolinsky) [inline alica_engine/include/engine/RunningPlan.h]: amISuccessful
^

File: alica_engine/include/engine/RunningPlan.h [hunk 2]
```diff
@@ -194,6 +194,8 @@ class RunningPlan : public std::enable_shared_from_this<RunningPlan>
...
+    bool AmISuccesfulInAnyChild() const;
```
Comment (abolinsky) [inline alica_engine/include/engine/RunningPlan.h]: amISuccessfulInAnyChild
^________^

**Review by abolinsky — CHANGES_REQUESTED** [review]
see comments

File: alica_engine/include/engine/Assignment.h [hunk 3]
```diff
@@ -87,6 +87,7 @@ class Assignment
     bool isValid() const;
     bool isSuccessful() const;
     bool isAnyTaskSuccessful() const;
+    bool isAgentSuccessful(AgentIDConstPtr id, const EntryPoint* ep) const;
```
Comment (abhi1287) [inline alica_engine/include/engine/Assignment.h:90]: const AgentIDConstPtr& id

Comment (abolinsky) [reply alica_engine/include/engine/Assignment.h:90]: AgentIDConstPtr is already a wrapper around a pointer to const type, so it should be passed by value.

File: alica_engine/include/engine/RunningPlan.h (hunk 1 above)
Comment (hendrik-skubch) [reply alica_engine/include/engine/RunningPlan.h]: of course!

File: alica_engine/include/engine/RunningPlan.h (hunk 2 above)
Comment (hendrik-skubch) [reply alica_engine/include/engine/RunningPlan.h]: +1

File: alica_engine/include/engine/Assignment.h (hunk 3 above)
Comment (hendrik-skubch) [reply alica_engine/include/engine/Assignment.h:90]: yup, what he said

Comment (abhi1287) [reply alica_engine/include/engine/Assignment.h:90]: missed that

**Review by abhi1287 — APPROVED** [review]

File: alica_engine/include/engine/RunningPlan.h [hunk 4]
```diff
@@ -194,6 +194,8 @@ class RunningPlan : public std::enable_shared_from_this<RunningPlan>
     bool isAnyChildStatus(PlanStatus ps) const;
     bool areAllChildrenStatus(PlanStatus ps) const;
     bool isAnyChildTaskSuccessful() const;
+    bool amISuccessful() const;
+    bool amISuccesfulInAnyChild() const;
```
Comment (abolinsky) [inline alica_engine/include/engine/RunningPlan.h]: amISuccessfulInAnyChild

**Review by abolinsky — CHANGES_REQUESTED** [review]

File: alica_engine/include/engine/RunningPlan.h (hunk 4 above)
Comment (hendrik-skubch) [reply alica_engine/include/engine/RunningPlan.h]: duh, fixed

**Review by abolinsky — APPROVED** [review]

**Pull request merged**